File: backend/scripts/convert_rtsp_format.js
const fs = require('fs');
const path = require('path');

const dir = path.join(__dirname);
const inputFile = path.join(dir, 'rtsp กล้องเก่า ม.json');
const outputFile = path.join(dir, 'rtsp กล้องเก่า ม_new.json');

let content = fs.readFileSync(inputFile, 'utf8').trim();

// Wrap in array if not already
if (!content.startsWith('[')) {
  content = '[' + content + ']';
}
// Remove trailing commas before ]
content = content.replace(/,(\s*\])/g, '$1');

const cameras = JSON.parse(content);

function parseWKT(wkt) {
  if (!wkt) return { lat: '', lon: '' };
  const m = wkt.match(/POINT\s*\(([0-9.]+)\s+([0-9.]+)\)/);
  if (m) {
    return {
      lat: parseFloat(m[2]).toFixed(4) + 'N',
      lon: parseFloat(m[1]).toFixed(4) + 'E'
    };
  }
  return { lat: '', lon: '' };
}

function buildRTSP(ip, rtsp) {
  if (rtsp) {
    // Unescape \/ -> /
    return rtsp.replace(/\\\//g, '/');
  }
  if (ip) {
    return `rtsp://mfustream:mediamfu2025@${ip}/Streaming/Channels/101/`;
  }
  return '';
}

const result = cameras.map((cam, i) => {
  const { lat, lon } = parseWKT(cam.WKT || '');
  return {
    'NO': String(i + 1),
    'IP ADDRESS': cam.ip || '',
    'CAMERA NAME_NEW': cam.name || '',
    'BUILDING': '',
    'FLOOR': '',
    'POSITION': '',
    'Latitude': lat,
    'Longtitude': lon,
    'Location': cam.name || '',
    'enable rtsp': '',
    'ANPR&PTZ RTSP': buildRTSP(cam.ip, cam.rtsp),
    'PTZ': '',
    '': ''
  };
});

fs.writeFileSync(outputFile, JSON.stringify(result, null, 2), 'utf8');
console.log(`Done! Converted ${result.length} cameras -> ${outputFile}`);

File: backend/scripts/convert_rtsp_format.py
import json
import re

input_file = r"c:\Users\Classroom\Documents\GitHub\CCTVmfu\backend\scripts\rtsp กล้องเก่า ม.json"
output_file = r"c:\Users\Classroom\Documents\GitHub\CCTVmfu\backend\scripts\rtsp กล้องเก่า ม_new.json"

with open(input_file, "r", encoding="utf-8") as f:
    content = f.read().strip()

# The file might not be a valid JSON array, wrap if needed
if not content.startswith("["):
    content = "[" + content + "]"

# Fix trailing commas before ]
content = re.sub(r",\s*\]", "]", content)

cameras = json.loads(content)

def parse_wkt_point(wkt):
    """Parse WKT POINT (lon lat) -> (lat_str, lon_str)"""
    match = re.match(r"POINT\s*\(([0-9.]+)\s+([0-9.]+)\)", wkt or "")
    if match:
        lon = float(match.group(1))
        lat = float(match.group(2))
        return f"{lat:.4f}N", f"{lon:.4f}E"
    return "", ""

def build_rtsp_url(ip, rtsp_raw):
    """Build clean RTSP URL from raw field."""
    if rtsp_raw:
        # Unescape \/ -> /
        return rtsp_raw.replace("\\/", "/")
    if ip:
        return f"rtsp://mfustream:mediamfu2025@{ip}/Streaming/Channels/101/"
    return ""

result = []
for i, cam in enumerate(cameras, start=1):
    wkt = cam.get("WKT", "")
    name = cam.get("name", "")
    ip = cam.get("ip", "")
    rtsp_raw = cam.get("rtsp", "")

    lat_str, lon_str = parse_wkt_point(wkt)
    rtsp_url = build_rtsp_url(ip, rtsp_raw)

    new_entry = {
        "NO": str(i),
        "IP ADDRESS": ip,
        "CAMERA NAME_NEW": name,
        "BUILDING": "",
        "FLOOR": "",
        "POSITION": "",
        "Latitude": lat_str,
        "Longtitude": lon_str,
        "Location": name,
        "enable rtsp": "",
        "ANPR&PTZ RTSP": rtsp_url,
        "PTZ": "",
        "": ""
    }
    result.append(new_entry)

with open(output_file, "w", encoding="utf-8") as f:
    json.dump(result, f, ensure_ascii=False, indent=2)

print(f"Done! Converted {len(result)} cameras -> {output_file}")
